build/lifecycle.go

```diff
@@ -76,7 +76,8 @@ func (l *Lifecycle) Execute(ctx context.Context, opts LifecycleOptions) error {
 		l.logger.Debugf("Build cache %s cleared", style.Symbol(buildCache.Name()))
 	}
 
-	if lifecycleVersion := l.builder.GetLifecycleVersion(); lifecycleVersion == nil {
+	lifecycleVersion := l.builder.GetLifecycleVersion()
+	if lifecycleVersion == nil {
 		l.logger.Debug("Warning: lifecycle version unknown")
```

Comment:
Now that we want to treat an unknown lifecycle version like the default lifecycle, we should probably set lifecycleVersion to the default in this case so that it is easy to be consistent.